### PR TITLE
Fix #835: (follow-up) Add missing configuration required for KubernetesResourceTask

### LINKED ISSUE #835: KubernetesResourceTask: KubernetesExtension has required configuration options
## Description

Child of #31 

KubernetesResourceTask can be configured

## AC
- [x] KubernetesExtension has a `getPlatformMode()` method that returns `kubernetes`
- [x] OpenShiftExtension extends `getPlatformMode()` method and returns `openshift`
- [x] KubernetesExtension has a `resourceTargetDirectory` field (`Property<File>`) to specify the YAML resource generation output
  - Defaults to `${build.classes.directory}/META-INF/jkube`
- [x] KubernetesExtension has a `resourceSourceDirectory` field (`Property<File>`) to specify the directory where source fragments can be found
  - Defaults to `${basedir}/src/main/jkube`
- [x] KubernetesExtension has a `resourceEnvironment` field (`Property<String>`) to specify the environment directory (dev, prod, etc.) where source fragments can be found (if not blank)
- [x] KubernetesExtension has a `useProjectClasspath` field (`Property<Boolean>`) to specify if JKube should search for additional enrichers/generators in the project's classpath
  - Defaults to `false`
- [x] KubernetesExtension has a `workDirectory` field (`Property<File>`) to specify the directory JKube uses for temporary file generation and work.
- [x] KubernetesExtension has a `skipResourceValidation` field (`Property<Boolean>`) that enables/disable ResourceValidation
  - Defaults to `false`
- [x] KubernetesExtension has a `failOnValidationError` field (`Property<Boolean>`) that causes resource generation to fail in case the generated resources are invalid
  - Defaults to `false`
- [x] KubernetesExtension has a `resources` field (`ResourceConfig`)
- [x] KubernetesExtension has a `profile` field (`Property<String>`) to enable the usage of a specific profile
- [x] KubernetesExtension has a `enricher` field (`ProcessorConfig`) to allow the configuration of Enrichers
- [x] KubernetesExtension has a `generator` field (`ProcessorConfig`) to allow the configuration of Generators
- [x] KubernetesExtension has a `mappings` field (`List<MappingConfig>`) that allows overriding the default fileName Resource type mappings
- [x] KubernetesExtension has a `namespace` field (`Property<String>`) that overrides ResourceConfig#namespace if present (not blank)
- [x] KubernetesExtension has a `resourceFileType` field (`Property<ResourceFileType>`) to allow the type of resource files to be generated (YAML / JSON)
  - Defaults to `yaml`
- [x] KubernetesExtension has a `mergeWithDekorate` field (`Property<Boolean>`) to enable the Dekorate resource merge experimental feature
  - Defaults to `false`
- [x] KubernetesExtension has a `interpolateTemplateParameters` field (`Property<Boolean>`) to allow enabling/disabling template placeholder interpolation
  - Defaults to `true`
- [x] KubernetesResourceTask overrides initJKubeServiceHubBuilder to provide a resourceService configuration
- [x] Create a profiles-default.yml file in `gradle-plugin/kubernets/src/main/resources/META-INF/jkube` with the same content found in the KubernetesMavenPlugin equivalent
- [x] Create a profiles-default.yml file in `gradle-plugin/openshift/src/main/resources/META-INF/jkube` with the same content found in the OpenShiftMavenPlugin equivalent

### PR DESCRIPTION
**Pull request by rohanKanojia**
## Description
Fix #835 

Add missing configuration left in #852

- Create a profiles-default.yml file in `gradle-plugin/openshift/src/main/resources/META-INF/jkube` with the same content found in the OpenShiftMavenPlugin equivalent
- Create a profiles-default.yml file in `gradle-plugin/kubernets/src/main/resources/META-INF/jkube` with the same content found in the KubernetesMavenPlugin equivalent
- KubernetesResourceTask overrides initJKubeServiceHubBuilder to provide a resourceService configuration

Signed-off-by: Rohan Kumar <rohaan@redhat.com>

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [X] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [X] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [X] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have implemented unit tests to cover my changes
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: gradle-plugin/kubernetes/src/main/java/org/eclipse/jkube/gradle/plugin/task/AbstractJKubeTask.java [hunk 1]
```diff
@@ -49,6 +50,7 @@
   protected JKubeServiceHub jKubeServiceHub;
   protected static final String DOCKER_BUILD_TIMESTAMP = "docker/build.timestamp";
   protected List<ImageConfiguration> resolvedImages;
+  protected File realResourceDir;
```
Comment (manusa) [inline gradle-plugin/kubernetes/src/main/java/org/eclipse/jkube/gradle/plugin/task/AbstractJKubeTask.java:53]: nit: `realResourceDir`doesn't provide much meaning. Either `applicableResourceDir` or `resolvedResourceDir` (for consistency with the other class variable) make more sense

Comment (manusa) [reply gradle-plugin/kubernetes/src/main/java/org/eclipse/jkube/gradle/plugin/task/AbstractJKubeTask.java:53]: Where is this set?

File: gradle-plugin/kubernetes/src/main/java/org/eclipse/jkube/gradle/plugin/task/KubernetesResourceTask.java [hunk 2]
```diff
@@ -16,10 +16,42 @@
...
 @SuppressWarnings("CdiInjectionPointsInspection")
 public class KubernetesResourceTask extends AbstractJKubeTask {
 
+  @Override
+  protected JKubeServiceHub.JKubeServiceHubBuilder initJKubeServiceHubBuilder() {
+    JKubeServiceHub.JKubeServiceHubBuilder builder = super.initJKubeServiceHubBuilder();
+    realResourceDir = ResourceUtil.getFinalResourceDir(
+      kubernetesExtension.getResourceSourceDirectory().getOrElse(new File(javaProject.getBaseDirectory(), "src/main/jkube")),
+      kubernetesExtension.getResourceEnvironment().getOrNull()
+    );
```
Comment (manusa) [inline gradle-plugin/kubernetes/src/main/java/org/eclipse/jkube/gradle/plugin/task/KubernetesResourceTask.java:37]: I the field is set here, it makes no sense to declare it in the parent class (https://github.com/eclipse/jkube/pull/853/files#r688239259)

Also, why not just creating a getApplicableResourceDirectory() in KubernetesExtension that performs the resolution logic?

Comment (rohanKanojia) [reply gradle-plugin/kubernetes/src/main/java/org/eclipse/jkube/gradle/plugin/task/KubernetesResourceTask.java:37]: realResourceDir is required for extracting enricher config, I have only ported sections related to #835 . You can see this field is used in AbstractJKubeTask `extractEnricherConfig()` in ResourceTask PR #834

Comment (manusa) [reply gradle-plugin/kubernetes/src/main/java/org/eclipse/jkube/gradle/plugin/task/KubernetesResourceTask.java:37]: I really don't get your point, if the field is used globally, it should be initialized in the parent class too, not only declared there.

Anyway, just follow the second suggestion:

> Also, why not just creating a getApplicableResourceDirectory() in KubernetesExtension that performs the resolution logic?

File: gradle-plugin/kubernetes/src/main/java/org/eclipse/jkube/gradle/plugin/KubernetesExtension.java [hunk 3]
```diff
@@ -288,4 +292,39 @@ public void mapping(Closure<?> closure) {
...
+  public boolean getApplicableFailOnValidationError() {
+    return getFailOnValidationError().getOrElse(false);
+  }
+
+  public boolean getApplicableMergeWithDekorate() {
+    return getMergeWithDekorate().getOrElse(false);
+  }
+
+  public boolean getApplicableInterpolateTemplateParameters() {
+    return getInterpolateTemplateParameters().getOrElse(false);
```
Comment (manusa) [inline gradle-plugin/kubernetes/src/main/java/org/eclipse/jkube/gradle/plugin/KubernetesExtension.java:316]: Should default to `true`

**Review by manusa — APPROVED** [review]

**Pull request merged**